### PR TITLE
Add a test suite

### PR DESCRIPTION
**Pull request by mato**
TL;DR: `sudo tests/setup-tests.sh && sudo tests/run-tests.sh`.

Finally, a test suite that runs end to end tests using the standalone test programs in `tests/`. See tests/README.md for usage.

The test harness is a rather gnarly shell script, this is the simplest I could get to that has the basic functionality we need. There are more interesting things that can be done with automated tests, but those will probably need a test harness written in something better than shell.

Please review. 

@hannesm There is a race condition _somewhere_ in the innards of `solo5-run-virtio.sh` for Bhyve which may cause random hard hangs (not sure why `timeout` is not doing the right thing in this case). Will debug in due course.

Part of #82, (1) "end to end CI".

### CODE REVIEW

File: tests/test_ping_serve/test_ping_serve.c [hunk 1]
```diff
@@ -317,6 +319,10 @@ static void ping_serve(int verbose)
         if (solo5_net_write_sync(buf, len) == -1)
             puts("Write error\n");
 
+        received++;
+        if (received == 100000)
```
Comment (ricarkol) [inline tests/test_ping_serve/test_ping_serve.c]: Can you add a message here saying that we are exiting because of this limit please.

It would be nice to pass this number as a parameter as I usually test virtio-net by letting this run for a while. Do you mind if i create a PR afterwards making this an arg? (unless you want to do it as part of this PR)

**Comment by mato** [comment]
@ricarkol I've implemented argument passing to the test unikernels in 039f206, does this address your comments?

**Comment by mato** [comment]
@ricarkol I've implemented argument passing to the test unikernels in 039f206, does this address your comments?
